### PR TITLE
Tables filtering

### PR DESCRIPTION
**Pull request by MariaDolotova**

### CODE REVIEW

File: UI/src/app/components/comfy/comfy.component.ts [hunk 1]
```diff
@@ -347,22 +347,23 @@ export class ComfyComponent extends BaseComponent implements OnInit, AfterViewIn
   }
 
   findTables(selectedSourceColumns: string[]): void {
+    if (selectedSourceColumns.length) {
     const indexes = {};
```
Comment (costebano) [inline UI/src/app/components/comfy/comfy.component.ts:351]: please fix indentation

File: UI/src/app/components/comfy/comfy.component.ts [hunk 2]
```diff
@@ -407,15 +408,18 @@ export class ComfyComponent extends BaseComponent implements OnInit, AfterViewIn
...
           break;
         }
         case 'target': {
           this.targetTableNames = uniq(Object.keys(this.targetConfig)).filter(filterByName);
           break;
         }
         case 'source-column': {
-          this.sourceRows = this.sourceRows.filter(row => filterByName(row.name));
+          this.sourceRows = uniqBy(this.data.source.
+          map(table => table.rows).
```
Comment (etokareva) [inline UI/src/app/components/comfy/comfy.component.ts:420]: you don't need to have three operations there. You already use reduce, you can get rows for table and apply filtering inside of it

File: UI/src/app/components/comfy/comfy.component.ts (hunk 1 above)
Comment (MariaDolotova) [reply UI/src/app/components/comfy/comfy.component.ts]: Done

File: UI/src/app/components/comfy/comfy.component.ts (hunk 2 above)
Comment (MariaDolotova) [reply UI/src/app/components/comfy/comfy.component.ts]: Done

File: UI/src/app/components/comfy/comfy.component.ts [hunk 3]
```diff
@@ -347,22 +347,23 @@ export class ComfyComponent extends BaseComponent implements OnInit, AfterViewIn
...
+        indexes[table.name] = tableIncludesColumns(rowNames, selectedSourceColumns);
+      });
 
-    this.highlitedtables = Object.keys(indexes).filter(tableName => {
-      return (
-        indexes[tableName].length > 0 &&
-        (indexes[tableName].findIndex(idx => idx > -1) > -1)
-      );
+      this.highlitedtables = Object.keys(indexes).filter(tableName => {
+      return indexes[tableName];
```
Comment (etokareva) [inline UI/src/app/components/comfy/comfy.component.ts:360]: move this return on line above

**Pull request merged**